libvast/builtins/connectors/file.cpp

```diff
@@ -0,0 +1,262 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/concept/parseable/to.hpp>
+#include <vast/concept/parseable/vast/data.hpp>
+#include <vast/detail/fdinbuf.hpp>
+#include <vast/detail/file_path_to_parser.hpp>
+#include <vast/detail/posix.hpp>
+#include <vast/detail/string.hpp>
+#include <vast/logger.hpp>
+#include <vast/plugin.hpp>
+
+#include <caf/error.hpp>
+
+#include <fcntl.h>
+#include <filesystem>
+#include <memory>
+#include <string_view>
+#include <unistd.h>
+
+namespace vast::plugins::file {
+
+class plugin : public virtual loader_plugin {
+public:
+  static constexpr inline auto max_chunk_size = size_t{16384};
+  static constexpr inline auto stdin_path = "-";
+
+  auto
+  make_loader(std::span<std::string const> args, operator_control_plane&) const
+    -> caf::expected<generator<chunk_ptr>> override {
+    auto read_timeout = read_timeout_;
+    auto path = std::string{};
+    auto file_type = std::filesystem::file_type::regular;
```

Comment:
This is not really the file type (as in `std::filesystem::file_type`, but rather a `auto is_socket = false` (this would also remove the unreachable `default` case).

Reply:
Yes, this would make things simpler. I replaced the `file_type` with `is_socket` in https://github.com/tenzir/vast/pull/3088/commits/a3b046c402783025a72fa735ac583f17f6598ba3